src/gui/src/SongEditor/SongEditor.cpp

```diff
@@ -2357,7 +2381,9 @@ void SongEditorPositionRuler::paintEvent( QPaintEvent *ev )
 
 void SongEditorPositionRuler::updatePosition()
 {
-	HydrogenApp::get_instance()->getSongEditorPanel()->updateTimelineUsage();
+	SongEditorPanel *pSEPanel = HydrogenApp::get_instance()->getSongEditorPanel();
+	pSEPanel->updateTimelineUsage();
+	pSEPanel->getSongEditor()->update();
```

Comment:
This runs constantly even when the audio engine isn't running. We should probably change that, but until now it's not been a problem -- the ruler is very fast to update() because it's small and has a pixmap as backing store. But the song editor is larger and may render more components each paint event (keyboard cursor, selection / move) so would be better to avoid redrawing it.

Keeping a static fLastPos in the SongEditorPanel and making `update()` conditional on that differing from the current pos would do the job fine :)